### ISSUE TITLE
Add support for DNS-over-QUIC to systemd-resolved

### ISSUE THREAD

**Issue by benoitjpnet** [systemd/systemd#23770]
**Is your feature request related to a problem? Please describe.**
DNS-over-QUIC (DoQ) is not supported.

**Describe the solution you'd like**
DoQ implemented.

The long-awaited DNS-over-QUIC protocol has finally graduated from a draft to being a real standard, [RFC 9250](https://datatracker.ietf.org/doc/html/rfc9250).

Example of software compatible DoQ: 
- [AdGuardHome](https://github.com/AdguardTeam/AdGuardHome), can resolve DoQ and be a DoQ resolver.

**The systemd version you checked that didn't have the feature you are asking for**

251.2

Somehow related (especially discussions): #8639

**Comment by rhymeswithmogul** [comment]
> This means that HTTP/3 has also ushered in an official version?

Sure did.  [RFC 9114](https://www.rfc-editor.org/rfc/rfc9114.html) just got released.

**Comment by codewiz** [comment]
This is the HTTP/3 stack that was used to build DoH3 for Android last year: https://github.com/cloudflare/quiche

It's a solid, actively maintained, portable, secure codebase with a small library footprint (given the complexity of QUIC and H3), and I highly recommend it.

Quiche does not support legacy HTTP/1 and HTTP/2, but since those were TCP-based, there would be no advantage over DoT (in fact, they have higher protocol overhead).

**Comment by nikelborm** [comment]
Hi, any updates on this?

**Comment by Bad3r** [comment]
… it's unfortunate that there is no update on this in the past 3 years.

**Comment by serhii-nakon** [comment]
yep I think it way better than regular DoT or DoH that has issues like dead TCP sessions that need to be handled manually in opposite to DoQ that make both encrypted and over UDP with lover latency and without dead TCP sessions

**Comment by tolikzodrot** [comment]
RFC 9250 is now 4 yo and DoQ is seeing real adoption on the resolver side (Quad9, Adguard, NextDNS, Control D all support it). The 0-RTT resumption and lack of TCP head-of-line blocking make it a meaningfully faster transport than DoT, not just marginally. Any update on whether this is actually planned, or is it just not going to happen?

**Comment by serhii-nakon** [comment]
I not sure but make more sense to add DnsOverHttps(3) and it will be way better becaus it already use quic and more providers support this then raw quic
